libs/feature/editor/src/lib/components/record-form/form-field/form-field-spatial-extent/form-field-spatial-extent.component.ts

```diff
@@ -1,10 +1,146 @@
+    CommonModule,
+    UiWidgetsModule,
+    AutocompleteComponent,
+    FeatureMapModule,
+    SwitchToggleComponent,
+    GenericFormFieldKeywordsComponent,
+    FormFieldMapContainerComponent,
+  ],
 })
-export class FormFieldSpatialExtentComponent {}
+export class FormFieldSpatialExtentComponent implements OnInit {
+  @Input() placeKeywords: Keyword[]
+  @Input() spatialExtents: DatasetSpatialExtent[]
+
+  @Output() placeKeywordChange: EventEmitter<Keyword[]> = new EventEmitter()
+  @Output() spatialExtentsChange: EventEmitter<DatasetSpatialExtent[]> =
+    new EventEmitter()
+
+  keywordsLinkedToExtents = new Array<{
+    [key: string]: {
+      placeKeyword?: Keyword
+      spatialExtents?: DatasetSpatialExtent
+    }
+  }>()
+
+  ngOnInit(): void {
+    this.linkPlaceKeywordsToSpatialExtents(
+      this.placeKeywords,
+      this.spatialExtents
+    )
+    this.linkSpatialExtentsToPlaceKeywords(
+      this.spatialExtents,
+      this.placeKeywords
+    )
+  }
+
+  linkPlaceKeywordsToSpatialExtents(
+    placeKeywords: Keyword[],
+    spatialExtents: DatasetSpatialExtent[]
+  ) {
+    placeKeywords.forEach((keyword) => {
+      const coordsBbox = this.transformCoordsToBbox(
+        keyword.coords?.coordEast,
+        keyword.coords?.coordNorth,
+        keyword.coords?.coordSouth,
+        keyword.coords?.coordWest
+      )
+      const bbox = spatialExtents.find(
+        (extent) => extent?.description === keyword?.key
+      )?.bbox
+
+      const geometries = spatialExtents.find(
+        (extent) => extent?.description === keyword?.key
+      )?.geometries
+
+      this.keywordsLinkedToExtents[keyword?.key] = {
+        placeKeyword: keyword,
+        spatialExtents: {
+          bbox: bbox?.length >= 0 ? bbox : coordsBbox,
+          geometries: geometries,
+          description: keyword.label,
+        },
+      }
+    })
+
+    this.placeKeywords = placeKeywords
+  }
+
+  transformCoordsToBbox(
+    coordEast: string,
+    coordNorth: string,
+    coordSouth: string,
+    coordWest: string
+  ): [number, number, number, number] {
+    return [coordWest, coordSouth, coordEast, coordNorth].map((coord) =>
+      parseFloat(coord)
+    ) as [number, number, number, number]
+  }
+
+  linkSpatialExtentsToPlaceKeywords(
+    spatialExtents: DatasetSpatialExtent[],
+    placeKeywords: Keyword[]
+  ) {
+    spatialExtents.forEach((extent) => {
+      if (this.keywordsLinkedToExtents[extent.description]) {
+        return
+      } else {
+        this.keywordsLinkedToExtents[extent.description] = {
+          spatialExtents: extent,
+          placeKeyword: {
+            key: extent.description,
+            label: 'Unknown location',
+            type: 'place',
+          },
+        }
+      }
+    })
+
+    const missingPlaces = Object.keys(this.keywordsLinkedToExtents).filter(
+      (uri) => !placeKeywords.some(({ key: id }) => uri === id)
+    )
+
+    missingPlaces.forEach((missingPlace) => {
+      this.placeKeywords.push(
```

Comment:
I don't think we fix inconsistencies in a record if there are more spatialExtents than keywords. This link between place keywords and spatial extents is our custom way of doing things, not any standard.

We will push new place keywords and new spatial extents as the user adds them from our interface, but we will not push from what we've read from the record.